### PR TITLE
Upgrade to ton v13

### PR DESCRIPTION
**Pull request by hrmon**

### CODE REVIEW

File: src/lib/useContractAddress.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ function useContractAddress() {
   const isAddressValid = validateAddress(contractAddress);
   const verifiedAddress = isAddressValid ? Address.parse(contractAddress!) : null;
 
-  const verifiedAddressStr = verifiedAddress?.toFriendly() ?? null;
+  const verifiedAddressStr = verifiedAddress?.toString() ?? null;
   const verifiedAddresssHex = verifiedAddress?.toString() ?? null;
```
Comment (shaharyakir) [inline src/lib/useContractAddress.ts:13]: this looks suspicious - if we wanted one friendly representation (EQDerEP...) and one hex (0:deac43d3221d0e...) we need different calls

**Review by shaharyakir — COMMENTED** [review]
See comments, looks generally good, I will test functionality meanwhile.

File: src/lib/makeGetCall.ts [hunk 2]
```diff
@@ -1,38 +1,49 @@
...
       default:
-        throw new Error(`unknown type: ${type}, val: ${JSON.stringify(val)}`);
+        throw new Error(`unknown type: ${item.type}`);
     }
-  });
+  }
+  return parsedItems;
 }
 
 export async function makeGetCall<T>(
```
[the remark this answers is not in the document]
Comment (hrmon) [reply src/lib/makeGetCall.ts:49]: Yes, for calling get methods on submitted contracts (`Getters` component)

File: src/lib/useContractAddress.ts (hunk 1 above)
Comment (hrmon) [reply src/lib/useContractAddress.ts]: Right, `toString` is now `toRawString` in ton v13. Fixed it.

File: src/lib/makeGetCall.ts [hunk 3]
```diff
@@ -1,38 +1,49 @@
...
 
 function _prepareParams(params: any[] = []) {
-  return params.map((p) => {
+  const paramsTuple = new TupleBuilder();
+  params.forEach((p) => {
     if (p instanceof Cell) {
-      return ["tvm.Slice", p.toBoc({ idx: false }).toString("base64")];
-    } else if (p instanceof BN) {
-      return ["num", p.toString(10)];
+      paramsTuple.writeCell(p);
```
Comment (shaharyakir) [inline src/lib/makeGetCall.ts:16]: this should be `writeSlice`

File: src/lib/makeGetCall.ts [hunk 4]
```diff
@@ -1,38 +1,49 @@
...
+    } else if (typeof p === "bigint") {
+      paramsTuple.writeNumber(p);
     }
-
     throw new Error("unknown type!");
```
Comment (shaharyakir) [inline src/lib/makeGetCall.ts:20]: this always gets called

Comment (shaharyakir) [inline src/lib/makeGetCall.ts:20]: this always gets called

File: src/lib/makeGetCall.ts [hunk 5]
```diff
@@ -1,38 +1,49 @@
...
+    const item = stack.pop();
+    switch (item.type) {
+      case "int":
+        parsedItems.push((item as TupleItemInt).value);
       case "cell":
-        return Cell.fromBoc(Buffer.from(val.bytes, "base64"))[0];
-      case "list":
-        if (val.elements.length === 0) {
-          return null;
+        parsedItems.push((item as TupleItemCell).cell);
```
Comment (shaharyakir) [inline src/lib/makeGetCall.ts:35]: missing break in these switches

**Review by shaharyakir — COMMENTED** [review]
also make sure all `toNano(..)` are replaced with string params

File: src/lib/makeGetCall.ts (hunk 4 above)
Comment (shaharyakir) [reply src/lib/makeGetCall.ts:20]: still needs addressing after latest fixes (e.g. put in else clause)

**Pull request merged**